File: searching/binary_search.rb
def binary_search(arr, low, high, key)
  if low <= high
    mid = (high+low)/2
    if arr[mid] == key
      return mid
    elsif arr[mid] > key
      return binary_search(arr, low, mid-1, key)
    else
      return binary_search(arr, mid+1, high, key)
    end
  end
  return -1
end

arr = Array.new
puts "Enter the size of Array"
n = gets.chomp().to_i

puts ("Enter the " + n.to_s + " elements in the array.")
for i in 0..(n)-1
  x = gets.chomp()
  arr.push(x)
end

puts "Enter the element to be searched : "
key = gets.chomp()

result = binary_search(arr, 0, n-1, key)

result != -1 ? (puts "Element found at location " + result.to_s) : (puts "Element not found")

File: searching/Go/BinarySearch.go

package main
import "fmt"
 
func binarySearch(below int, up []int) bool {
 
    low := 0
    high := len(up) - 1
 
    for low <= high{
        median := (low + high) / 2
 
        if up[median] < below {
            low = median + 1
        }else{
            high = median - 1
        }
    }
 
    if low == len(up) || up[low] != below {
        return false
    }
	
    return true
}
 
func main(){
    items := []int{3, 15, 21, 36, 46, 67, 93, 110}
	var searchKey = 21
	if (binarySearch(searchKey, items) == true){
		fmt.Println(searchKey,"is available in the array")
	}else{
		fmt.Println(searchKey,"isn't available in the array")
	}

}